File: python/fib.py
def fib(n):
    if n <= 2:
        return n
    else:
        return (fib(n-1) + fib(n-2))


def fib2(n):
    a, b = 0, 1
    for i in range(n-1):
        a, b = b, a + b
    return b


def main():
    print(fib2(10))


if __name__ == '__main__':
    main()

File: c/fib.c
#include <stdlib.h>
#include <stdio.h>
#include <math.h>

int fib(int n)
{
    if(n <= 2)
        return n;
    else 
        return(fib(n-1)+ fib(n-2));
}


float fib2(int n)
{
    float fi = (1 + sqrt(5)) / 2;
    float f = (pow(fi, n) - pow(-fi, -n)) / (2 * fi - 1);
    return floor(f);
}

int main(int argc, char * argv[])
{
    int n = atoi(argv[1]);
    // printf("Number Fib:%d\n", fib(n));
    printf("Number Fib:%.0f\n",fib2(n));
    return 0;
}
